server/storage/storage_test.go

```diff
@@ -279,3 +281,129 @@ func TestLoadRegionsExceedRangeLimit(t *testing.T) {
 	}
 	re.NoError(failpoint.Disable("github.com/tikv/pd/server/storage/kv/withRangeLimit"))
 }
+
+const (
+	keyChars = "abcdefghijklmnopqrstuvwxyz"
+	keyLen   = 20
+)
+
+func generateKeys(size int) []string {
+	m := make(map[string]struct{}, size)
+	for len(m) < size {
+		k := make([]byte, keyLen)
+		for i := range k {
+			k[i] = keyChars[rand.Intn(len(keyChars))]
+		}
+		m[string(k)] = struct{}{}
+	}
+
+	v := make([]string, 0, size)
+	for k := range m {
+		v = append(v, k)
+	}
+	sort.Strings(v)
+	return v
+}
+
+func randomMerge(regions []*metapb.Region, n int) {
+	rand.Seed(6)
+	note := make(map[int]bool)
+	for i := 0; i < n; i++ {
+		if _, ok := note[i]; ok {
+			continue
+		}
+		note[i] = true
+		mergeIndex := rand.Intn(n - i) + i
+		for {
+			if _, ok := note[mergeIndex]; !ok {
+				if !ok {
+					break
+				}
+			}
+			mergeIndex = rand.Intn(n - i) + i
+		}
+		note[mergeIndex] = true
+		regions[i].EndKey = regions[mergeIndex].EndKey
+	}
+}
+
+func anotherSaveRegions(lb *levelDBBackend, n int, merge bool) error {
+	keys := generateKeys(n)
+	regions := make([]*metapb.Region, 0, n)
+	for i := 0; i < n; i++ {
+		var region *metapb.Region
+		if i == 0 {
+			region = &metapb.Region{
+				StartKey: []byte("aaaaaaaaaaaaaaaaaaaa"),
+				EndKey: []byte(keys[i]),
+			}
+		} else {
+			region = &metapb.Region{
+				StartKey: []byte(keys[i-1]),
+				EndKey: []byte(keys[i]),
+			}
+		}
+		regions = append(regions, region)
+	}
+	if merge {
+		randomMerge(regions, n)
+	}
+
+	for _, region := range regions {
+		err := lb.SaveRegion(region)
+		if err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+func benchmarkLoadRegions(n int, b *testing.B, merge bool) {
+	ctx := context.Background()
+	dir := b.TempDir()
+	lb, err := newLevelDBBackend(ctx, dir, nil)
+	if err != nil {
+		b.Fatal(err)
+	}
+	err = anotherSaveRegions(lb, n, merge)
+	if err != nil {
+		b.Fatal(err)
+	}
+	cluster := core.NewBasicCluster()
+	defer func() {
+		err = lb.Close()
+		if err != nil {
+			b.Fatal(err)
+		}
+	}()
+
+	b.StartTimer()
+	err = lb.LoadRegions(ctx, cluster.CheckAndPutRegion)
```

Comment:
Can you check the region numbers `LoadRegions` here? I wonder if the region is saved in cache and does not flush to the disk.